### PR TITLE
Usability review items #3

### LINKED ISSUE #1176: Usability issues: episode 3
### Internal Order Headers

- [ ] Maximum MOS: cannot change it (stuck to `3 Months`). 
- [x] Requested to Suggested button: use the mSupply mobile wording instead: `Use Suggested Quantities`
- [ ] `Requested to Suggested` button reset Requested to 0 and does not copy suggested quantities. 
- [x] `Hide stock over minimum` toggle: if all items are over minimum MOS, then table says "No items". This could be confusing cause technically there are items listed in the order.  

### Order modal (charts):

- [x] In the modal we say `order quantity` but in the list we say `Requested Quantity`, I would go for consistency and use Requested Quantity on both. 

### Order lines table

- [x] new order for the internal order table: code > name > SOH > AMC > Target Stock > Suggested Quantity > Requested Quantity > Comment 
- [x] SoH (Est. Remaining): could be replaced by SOH (MOS), maybe just 1 decimal ? 

## Documentation
- Update all docs links to use specific locations if available
  - [x] Dashboard: https://docs.msupply.foundation/docs/dashboard/
  - [x] Outbound shipments: https://docs.msupply.foundation/docs/distribution/outbound-shipments/
  - [x] Requisitions: https://docs.msupply.foundation/docs/distribution/requisitions/
  - [x] Customers: https://docs.msupply.foundation/docs/distribution/customers/
  - [x] Inbound shipments: https://docs.msupply.foundation/docs/replenishment/inbound-shipments/
  - [x] Internal orders: https://docs.msupply.foundation/docs/replenishment/internal-orders/
  - [x] Suppliers: https://docs.msupply.foundation/docs/replenishment/suppliers/
  - [x] Items: https://docs.msupply.foundation/docs/catalogue/items/
  - [x] Master lists: https://docs.msupply.foundation/docs/catalogue/master-list/
  - [x] Locations: https://docs.msupply.foundation/docs/inventory/locations/
  - [x] Stock: https://docs.msupply.foundation/docs/inventory/stock-view/
  - [x] Stocktakes: https://docs.msupply.foundation/docs/inventory/stock-takes/

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1176 

An additional item which we worked on during this one: add the requested quantity to the stock evolution chart to show what effect your order will have on stock levels.

### CODE REVIEW

File: packages/common/src/ui/layout/tables/hooks/useColumns/useColumns.test.tsx [hunk 1]
```diff
@@ -17,8 +17,8 @@ describe('useColumns', () => {
       sortInverted: false,
       sortDescFirst: false,
       align: ColumnAlign.Left,
-      width: 100,
-      minWidth: 100,
+      width: undefined,
```
Comment (mark-prins) [inline packages/common/src/ui/layout/tables/hooks/useColumns/useColumns.test.tsx:20]: offroad - fixing tests

File: packages/host/src/components/AuthenticationAlert.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { useToggle } from '@common/hooks';
-import { AppRoute } from 'packages/config/src';
+import { AppRoute } from '@openmsupply-client/config';
```
Comment (mark-prins) [inline packages/host/src/components/AuthenticationAlert.tsx:2]: updating to allow tests to run - they barf on the `packages/` imports

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/ConsumptionHistory.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ export const ConsumptionHistory: React.FC<ConsumptionHistoryProps> = ({
           ? t('label.consumption')
           : t('label.amc');
         return [value, label];
-      case 'amc':
+      case 'averageMonthlyConsumption':
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/ConsumptionHistory.tsx:46]: the props had been renamed in the api and I hadn't noticed

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockEvolution.tsx [hunk 4]
```diff
@@ -31,17 +35,20 @@ export const StockEvolution: React.FC<StockEvolutionProps> = ({ id }) => {
     switch (name) {
       case 'stockOnHand':
         return [value, t('label.stock-level')];
-      case 'min':
+      case 'minimumStockOnHand':
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockEvolution.tsx:38]: again, changed in api

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockEvolution.tsx [hunk 5]
```diff
@@ -31,17 +35,20 @@ export const StockEvolution: React.FC<StockEvolutionProps> = ({ id }) => {
...
+      case 'maximumStockOnHand':
         return [value, t('label.max')];
       default:
         return [value, name];
     }
   };
-
   if (!data || !data.stockEvolution) return null;
   const tooltipLabelFormatter = (date: string) => dateFormatter(date);
+  const chartData = data.stockEvolution.nodes.map(node => ({
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockEvolution.tsx:48]: was hoping to avoid this.. but, it does make the dynamic display of the chart pretty neat, when you adjust requested quantity

File: packages/requisitions/src/RequestRequisition/DetailView/columns.ts [hunk 6]
```diff
@@ -83,14 +74,36 @@ export const useRequestColumns = () => {
             itemStats;
 
           const monthsString = availableMonthsOfStockOnHand
-            ? `(${availableMonthsOfStockOnHand.toFixed(2)} ${t('label.months', {
-                count: availableMonthsOfStockOnHand,
-              })})`
+            ? `(${formatNumber.round(availableMonthsOfStockOnHand, 1)} ${t(
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/columns.ts:77]: `formatNumber` handles intl so will give `3.67` if language is english and `3,67` if you speak French

**Review by clemens-msupply — APPROVED** [review]
Looks good!

I haven't tested it though. Its in "Internal Order" right? but then I get: "Unable to calculate: No Average Monthly Consumption value"

File: packages/common/src/utils/quantities/quantities.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ export const QuantityUtils = {
     const total = amc * mos;
 
     // Subtract the available stock on hand
-    const suggested = total - Math.max(soh, 0);
+    const suggested = Math.round(total - Math.max(soh, 0));
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/common/src/utils/quantities/quantities.ts:14]: seems reasonable enough, yeah. there was a long discussion about decimal / fractional quantities.. mostly though we just have whole values for quantities

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 8]
```diff
@@ -109,19 +109,22 @@ export const RequestLineEditForm = ({
           {item && item?.unitName ? (
             <InfoRow label="Unit" value={item.unitName} />
           ) : null}
-          {item && item?.stats.averageMonthlyConsumption != null ? (
+          {!!draftLine?.itemStats.averageMonthlyConsumption ? (
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:112]: the former will also catch `!== undefined` and I generally find it safer than testing explicitly for `null`
Basically, it relates to a valid value.

**Comment by mark-prins** [comment]
> Its in "Internal Order" right? but then I get: "Unable to calculate: No Average Monthly Consumption value"

It is 😄  and yes, most have no stats values. You can switch to the Opua store and use https://demo-open.msupply.org/replenishment/internal-order/12 as an example

File: packages/common/src/utils/environment/EnvUtils.ts [hunk 9]
```diff
@@ -1,5 +1,62 @@
...
+      return { title: 'suppliers', docs: '/replenishment/suppliers/' };
+    case RegexUtils.includes('items', route):
+      return { title: 'items', docs: '/catalogue/items/' };
+    case RegexUtils.includes('master-lists', route):
+      return { title: 'master-lists', docs: '/catalogue/master-list/' };
+    case RegexUtils.includes('locations', route):
+      return { title: 'locations', docs: '/inventory/locations/' };
+    case RegexUtils.includes('stocktakes', route):
+      return { title: 'stocktakes', docs: '/inventory/stock-takes/' };
+    // Stocktakes needs to go before stock so matching is correct
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/common/src/utils/environment/EnvUtils.ts]: thanks @clemens-msupply that's very neat!
I've refactored and added a few tests in 70924e50

**Comment by mark-prins** [comment]
> I guess its expected behavior though, right?

yes, I think so. best to deal with whole pack numbers

**Comment by mark-prins** [comment]
> Not sure how much work it is but you could encode it into the type system:

Ah. thanks for the push. Types are already available, so I have changed from string to `keyof ConsumptionHistoryNode`

> Is it a bug that itemStats are integers?

The `item.stats` and `itemStats` should be consistent - I thought that Andrei was planning to remove one so haven't made an issue.. not sure. For AMC it was nicer, I think, to have a real value: showing `3.67` somehow feels more valid than `3` for a calculated value. maybe that's just me 🤷  

For one of the values, only `item` was available, so I have to use the stats property - but used `itemStats` as that was newer and I understood that was the preferred approach

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:112]: yes right thanks! `draftLine ?? draftLine?.itemStats.averageMonthlyConsumption ? (` should work as well though
